### PR TITLE
Fix reference implementation for ConcatFromSequence

### PR DESCRIPTION
**Pull request by xadupre**
### Description
Fix bug raised by issue #6366.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]

File: onnx/reference/ops/op_concat_from_sequence.py [hunk 1]
```diff
@@ -12,8 +12,12 @@
 
 def _concat_from_sequence(seq: list[Any], axis: int, new_axis: int = 0) -> np.ndarray:
     if new_axis == 1:
-        seq2 = [s[..., np.newaxis] for s in seq]
-        res = np.concatenate(seq2, axis=-1)
+        if axis == -1:
```
Comment (justinchuby) [inline onnx/reference/ops/op_concat_from_sequence.py:15]: Wondering why the special case for -1 is needed? Maybe add a comment?

Comment (xadupre) [reply onnx/reference/ops/op_concat_from_sequence.py:15]: expand_dims(axis=-1) did not have an obvious meaning (insert at the end or just before the last axis). So I distinguished the two cases.

**Comment by justinchuby** [comment]
@andife This PR was deployed to test pypi, and that seems unexpected?

File: onnx/reference/ops/op_concat_from_sequence.py (hunk 1 above)
Comment (gramalingam) [reply onnx/reference/ops/op_concat_from_sequence.py:15]: I still have the same question. I agree there is an ambiguity in the ONNX documentation ... but the ambiguity applies to values other than -1 also. Eg., -2 as well. 

I think that the spec means that the specified axis is interpreted with respect to the output's shape (not the input's shape). So, the specified axis in the output will be the newly inserted axis.

If I understand the spec of `np.expand_dims`, it is similar: https://numpy.org/doc/stable/reference/generated/numpy.expand_dims.html ... so, something seems off here.

Comment (justinchuby) [reply onnx/reference/ops/op_concat_from_sequence.py:15]: I think this still needs to be address? Created # to track

**Pull request merged**